File: src/main/kotlin/ps/api/module/user/dto/SignUpDto.kt
package ps.api.module.user.dto

import kotlinx.serialization.Serializable
import leight.client.sdk.annotation.TypeString
import leight.dto.AbstractDto
import leight.sdk.annotation.Module

@Serializable
@Module("shared/user")
data class SignUpDto(
	@TypeString
	val name: String,
	@TypeString
	val login: String,
	@TypeString
	val password: String,
) : AbstractDto()

File: src/PuffSmith/Api/Shared/User/Endpoint/Dto/SignUpDto.php
<?php
declare(strict_types=1);

namespace PuffSmith\Api\Shared\User\Endpoint\Dto;

use Edde\Dto\AbstractDto;

class SignUpDto extends AbstractDto {
	/**
	 * @var string
	 */
	public string $email;
	/**
	 * @var string
	 */
	public string $name;
	/**
	 * @var string
	 */
	public string $password;
}
